src/main/java/no/unit/nva/doi/CrossRefClient.java

```diff
@@ -39,15 +39,16 @@ public CrossRefClient(HttpClient httpClient) {
         this.httpClient = httpClient;
     }
 
-    public Optional<String> fetchDataForDoi(String doiIdentifier) throws URISyntaxException {
+    public Optional<FetchResult> fetchDataForDoi(String doiIdentifier) throws URISyntaxException {
```

Comment:
FetchResult contains the json object and the location from where it was fetched (Datacite, Crossref).

Reply:
Maybe write this comment in code instead?

Reply:
Added comment to code.